content/blog/2022-04-15-training-and-saving-models-on-a-dedicated-runner-part-2.md

```diff
@@ -0,0 +1,236 @@
+and not pushed to the repository.
+
+The other files created in `train.py` are still pushed to an experiment branch
+in GitHub. Afterwards a merge request is created.
+
+```yaml
+name: CML with DVC
+on: # Here we use two triggers: manually and daily at 08:00
+  workflow_dispatch:
+  schedule:
+    - cron: '0 8 * * *'
+jobs:
+  deploy-runner:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: iterative/setup-cml@v1
+      - name: Deploy runner on EC2
+        env:
+          REPO_TOKEN: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        run: |
+          cml runner \
+              --cloud=aws \
+              --cloud-region=eu-west \
+              --cloud-type=t2.micro \
+              --labels=cml-runner \
+              --single
+  train-model:
+    needs: deploy-runner
+    runs-on: [self-hosted, cml-runner]
+    timeout-minutes: 120 # 2h
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+      - uses: iterative/setup-cml@v1
+      - name: Train model
+        env:
+          REPO_TOKEN: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+          GDRIVE_CREDENTIALS_DATA: ${{ secrets.GDRIVE_CREDENTIALS_DATA }}
+        run: |
+          cml ci
+          pip install -r requirements.txt
+
+          python get_data.py
+          python train.py
+
+          # This automatically adds the model to your .gitignore
+          dvc add model/random_forest.joblib
+
+          # Connect to your DVC remote storage and push the model to there
+          dvc remote add -d -f myremote gdrive://${{ secrets.GOOGLE_DRIVE_URI }}
+          dvc remote modify myremote gdrive_use_service_account true
+
+          dvc push
+
+          # Create pull request for the remaining files
+          cml pr .
+
+          # Create CML report
+          cat model/metrics.txt > report.md
+          cml publish model/confusion_matrix.png --md >> report.md
+          cml send-comment --pr --update report.md
+```
+
+And that's it! We have broadly the same set-up as outlined in part 1 of this
+guide, but we no longer use our GitHub repository for storing our model.
+Instead, the model is now saved to Google Drive, which allows for much larger
+models.
+
+<admon type="tip">
+
+In a situation where we retrain our model daily based on the most recent data,
+it would make sense to also use DVC to keep track of the data used in each
+training. We could, for example, use our runner to import our training data from
+a table in our database and write both the data and the model to the DVC remote.
+This is beyond the scope of this guide, but
+[here you can find a tutorial for this](https://github.com/iterative/cml_dvc_case).
```

Comment:
Not sure "tutorial" is the right word for that.  Maybe "steps" or "instructions". Our audience loves repos.  Maybe better to say, "here you can find a repo with a template and steps for this."